compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -1312,6 +1312,9 @@ pub fn install_ice_hook(
         std::env::set_var("RUST_BACKTRACE", "full");
     }
 
+    // Lock stderr to prevent interleaving of concurrent panics.
+    let _guard = io::stderr().lock();
```

Comment:
Shouldn't you be taking the guard inside the panic handler? This function only sets up the panic handler

Reply:
Yeah.